File: SubArraySums.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        
        
        HashMap<Integer,Integer> h = new HashMap<>();
        
        
        
        int count=0;
        
        h.put(0,1);
        int currSum=0;
        
        for(int i =0;i<nums.length;i++)
        {
            currSum+=nums[i];
            int compliment = currSum - k;
            
            if(h.containsKey(compliment))
            {
                
               count+= h.get(compliment);
                
            }
            h.put(currSum,h.getOrDefault(currSum,0)+1);
          
        }
        return count;
    }
}

File: subarraysums.py
##Time Complexity : O(n)
##Space Complexity : O(n)
##Did this code successfully run on Leetcode : Yes

class Solution(object):
    def subarraySum(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        result = 0
        sum = 0
        presum = {0 : 1}
        for n in nums:
            sum +=n
            difference = sum - k
            result += presum.get(difference, 0)
            presum[sum] = 1 + presum.get(sum, 0)
            
        return result